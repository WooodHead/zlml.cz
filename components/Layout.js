// @flow

import Head from 'next/head';
import stylis from 'stylis';

import Wrapper from '../components/Wrapper';
import Colors from '../services/Colors';

type Props = {
  children?: Object,
  title?: string,
};

export default ({ children, title = 'This is the default title' }: Props) =>
  <div>
    <Head>
      <title>
        {title}
      </title>
      <meta charSet="utf-8" />
      <meta name="viewport" content="initial-scale=1.0, width=device-width" />
      <link rel="icon" href="/static/favicon.ico" />
      {/* Do not use styled-jsx here: https://github.com/zeit/next.js/issues/885 */}
      <style>
        {stylis(
          '',
          `
          body {
            font-family: sans-serif;
            font-weight: 300;
            font-size: 1.6rem;
            background-color: #f9f9f9;
            margin: 0;
            color: ${Colors.dark};
          }

          html {
            font-size: 62.5%;
          }

          p {
            hyphens: auto;
          }

          a,
          a:hover,
          a:focus {
            color: ${Colors.red};
          }

          a:visited {
            color: darkred;
          }

          a:hover,
          a:focus {
            text-decoration: underline;
          }

          a[href^="#error:"] {
            background: red;
            color: white;
          }
          `,
        )}
      </style>
    </Head>

    <Wrapper>
      {children}

      <footer>
        <p>
          By the way - this website is also{' '}
          <a href="https://github.com/mrtnzlml/zlml.cz">open-sourced</a>.
        </p>
      </footer>
    </Wrapper>
  </div>;
