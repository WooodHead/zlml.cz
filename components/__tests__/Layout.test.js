// @flow

import React from 'react';
import renderer from 'react-test-renderer';

import Layout from '../Layout';

it('renders correctly', () => {
  expect(renderer.create(<Layout>CONTENT</Layout>).toJSON()).toMatchSnapshot();
});
