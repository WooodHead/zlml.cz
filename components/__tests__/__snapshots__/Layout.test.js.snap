// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="wrapper"
    data-jsx={3382400623}
  >
    CONTENT
    <footer>
      <p>
        By the way - this website is also
         
        <a
          href="https://github.com/mrtnzlml/zlml.cz"
        >
          open-sourced
        </a>
        .
      </p>
    </footer>
  </div>
</div>
`;
